src/widget/date/datepicker-extended.js

```diff
@@ -89,7 +89,8 @@ class DatepickerExtended extends Widget {
 
         $fakeDateI.on('change paste', (e) => {
             let convertedValue = '';
-            let value = e.type === 'paste' ? getPasteData(e) : this.value;
+            let value =
+                e.type === 'paste' ? getPasteData(e.originalEvent) : this.value;
```

Comment:
We've had some discussion about generally moving away from jQuery. I've recently spotted some code where jQuery code has been partially migrated to regular DOM code, but still partially using jQuery APIs, causing the code not to run.

The tests would _hopefully_ catch that, but as a bit of extra precaution in case they didn't, I'd like to consider falling back to the event received in the callback like:

```suggestion
                e.type === 'paste' ? getPasteData(e.originalEvent ?? e) : this.value;
```

Reply:
Nice!  +1 on moving away from jQuery!  (Also TIL about [`??`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) so thanks for that!)